retry.go

```diff
@@ -0,0 +1,299 @@
+package relay
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/armon/relay/broker"
+)
+
+// retryBroker implements a resilient AMQP broker which automatically replaces
+// the underlying channel if an unexpected error is observed. Due to AMQP's
+// persistent connections, this becomes necessary to survive even minor network
+// hiccups which may temporarily sever the connection to the server.
+//
+// The retryBroker does not come without a price, however. Clients are still
+// responsible for deduplicating messages or handling them idempotently, as
+// duplications are a regular happening in the face of network errors due to
+// message redelivery provided by the protocol. It is also important to note
+// that acknowledgements - both ack and nack - cannot be retried across
+// connection resets. This stems from the fact that a given consumer's channel
+// cannot be revived and must be thrown away if the consumer is to be reused.
+type retryBroker struct {
+	broker   *relayBroker
+	attempts int
+	min      time.Duration
+	max      time.Duration
+}
+
+// RetryBroker returns a new retrying broker with the given settings.
+func (r *Relay) RetryBroker(attempts int, min, max time.Duration) *retryBroker {
```

Comment:
We should sanity check attempts is at least 1 and no more than 32 or something. Same with min / max validation.